context/app/static/js/components/workspaces/formHooks.ts

```diff
@@ -37,37 +16,32 @@ const errorHelper = {
     `You have selected ${protectedRows.length} protected datasets. Workspaces currently only supports published public datasets. To remove protected datasets from this workspace creation, click the “Remove Protected Datasets” button below or return to the previous screen to manually remove those datasets.`,
 };
 
-function useProtectedDatasetsForm() {
+function useWorkspacesProtectedDatasetsForm() {
   const { selectedRows, deselectRows } = useSelectableTableStore();
-  const { toastSuccessRemoveProtectedDatasets } = useWorkspaceToasts();
-  const protectedRows = useDatasetsAccessLevel(selectedRows.size > 0 ? [...selectedRows] : []).datasets;
-  const containsProtectedDataset = protectedRows.length > 0;
-
-  const reportedProtectedRows = useRef(false);
 
-  const errorMessages = [];
+  const trackEventHelper = useCallback((numProtectedRows: number) => {
+    trackEvent({
+      category: WorkspacesEventCategories.Workspaces,
+      action: 'Create Workspace / Protected datasets selected',
+      value: numProtectedRows,
+    });
+  }, []);
 
-  if (containsProtectedDataset) {
-    const protectedRowsError =
-      protectedRows.length === 1 ? errorHelper.protectedDataset : errorHelper.protectedDatasets;
-    errorMessages.push(protectedRowsError(protectedRows));
-    if (!reportedProtectedRows.current) {
-      reportedProtectedRows.current = true;
-      trackEvent({
-        category: WorkspacesEventCategories.Workspaces,
-        action: 'Create Workspace / Protected datasets selected',
-        value: protectedRows.length,
-      });
+  const protectedDatasetsErrorMessage = useCallback((protectedRows: DatasetAccessLevelHits) => {
+    if (protectedRows.length === 1) {
+      return errorHelper.protectedDataset(protectedRows);
     }
-  }
-  const protectedHubmapIds = protectedRows?.map((row) => row?._source?.hubmap_id).join(', ');
+    return errorHelper.protectedDatasets(protectedRows);
+  }, []);
 
-  const removeProtectedDatasets = useCallback(() => {
-    deselectRows(protectedRows.map((r) => r._id));
-    toastSuccessRemoveProtectedDatasets();
-  }, [deselectRows, protectedRows, toastSuccessRemoveProtectedDatasets]);
+  const { ...rest } = useProtectedDatasetsForm({
```

Comment:
We could probably return the result of the hook instead:

```suggestion
  return useProtectedDatasetsForm({
```